h2o-core/src/main/java/hex/grid/hyperspace/CartesianWalker.java

```diff
@@ -0,0 +1,125 @@
+package hex.grid.hyperspace;
```

Comment:
can you please point to the changes from the original impl?

Reply:
None besides this:

- The `Model` instance given to `hasNext` and `next` methods was allowed to be null. Now is Optional, which I believe shouts more clearly the model is optional. Both for users and future implementors.
- New methods for getting `n` intial models - those might be vended using a different strategy than calling `next` - especially they're guaranteed to know their surroundings and previous models.

Nothing else.

Reply:
Completely discarded the refactor.